src/library_signals.js

```diff
@@ -124,6 +124,19 @@ var funs = {
     ___setErrNo(ERRNO_CODES.EINTR);
     return -1;
   },
+#if ASSERTIONS
+  siglongjmp__deps: ['longjmp'],
+  siglongjmp: function(env, value) {
+    // We cannot wrap the sigsetjmp, but I hope that
+    // in most cases siglongjmp will be called later.
+
+    // siglongjmp can be called very many times, so don't flood the stderr.
+    Runtime.warnOnce("Calling longjmp() instead of siglongjmp()");
```

Comment:
I have implemented it like other signal-related functions that print a warning in an assertions-enabled build. The only drawback, I think, is the performance. When I first read the implementation of the `warnOnce` method, I thought it was terribly slow for a possibly hot code path. Then I had written some trivial test that calls `sigsetjmp` / `siglongjmp` many times in a loop, something similar to
```c
#include <setjmp.h>
#include <stdio.h>

int main() {
    sigjmp_buf env;
    volatile int x = 0;
    for(volatile int i = 0; i < 100000; ++i) {
        if(sigsetjmp(env, 1) == 0) {
            x += 1;
            siglongjmp(env, 1);
        }
    }   
    fprintf(stderr, "x = %d\n", x);
    return 0;
}
```
And, surprisingly, got negligible difference between `-O0` and `-O3` builds in `node` and Firefox and about 1.5x slowdown on Chromium. Maybe, `emcc` had optimized this somehow in the `-O0` build, though, so I'm not perfectly sure that the test was correct. In case your tests show that the `warnOnce` has significant overhead, I can revert this back to manual setting a flag and not relying on a set of strings.